From 3e9fe98cec6b3fb59b1b31183f570e3314a453be Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Fri, 1 Apr 2016 18:10:19 +0200
Subject: [PATCH 4/5] CMake: Add MacOSX Framework support

---
 CMakeLists.txt | 15 ++++++++++++++-
 1 file changed, 14 insertions(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1b22b5f..db000cf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -246,6 +246,9 @@ endforeach()
 set_option(VIDEO_COCOA         "Use Cocoa video driver" ${APPLE})
 set_option(DIRECTX             "Use DirectX for Windows audio/video" ${WINDOWS})
 set_option(RENDER_D3D          "Enable the Direct3D render driver" ${WINDOWS})
+if(APPLE)
+  set_option(BUILD_FRAMEWORK   "Build Framework bundle for OSX" OFF)
+endif()
 
 # TODO: We should (should we?) respect cmake's ${BUILD_SHARED_LIBS} flag here
 # The options below are for compatibility to configure's default behaviour.
@@ -1263,6 +1266,16 @@ if(SDL_SHARED)
       SOVERSION ${LT_REVISION}
       OUTPUT_NAME "SDL2")
   endif()
+  if(APPLE AND BUILD_FRAMEWORK)
+    set_target_properties(SDL2 PROPERTIES
+      FRAMEWORK TRUE
+      FRAMEWORK_VERSION ${SDL_VERSION}
+      MACOSX_FRAMEWORK_IDENTIFIER org.libsdl.SDL2
+      MACOSX_FRAMEWORK_SHORT_VERSION_STRING ${SDL_VERSION}
+      MACOSX_FRAMEWORK_BUNDLE_VERSION ${SDL_VERSION}
+      XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
+      OUTPUT_NAME "SDL2")
+  endif()
  set(_INSTALL_LIBS "SDL2" ${_INSTALL_LIBS})
  target_link_libraries(SDL2 ${EXTRA_LIBS} ${EXTRA_LDFLAGS})
 endif()
@@ -1290,7 +1303,7 @@ install(TARGETS ${_INSTALL_LIBS}
   ARCHIVE DESTINATION "lib${LIB_SUFFIX}"
   PUBLIC_HEADER DESTINATION include/SDL2)
 
-if(NOT WINDOWS OR CYGWIN)
+if((NOT WINDOWS OR CYGWIN) AND (NOT BUILD_FRAMEWORK))
   if(SDL_SHARED)
     install(CODE "
       execute_process(COMMAND ${CMAKE_COMMAND} -E create_symlink
