From f7cfaf6eff710f873a5fd2633c88c9b085450898 Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Sat, 9 Apr 2016 20:16:24 +0200
Subject: [PATCH 5/5] Add Mac OSX framework support

---
 cmake/libpython/CMakeLists.txt | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/cmake/libpython/CMakeLists.txt b/cmake/libpython/CMakeLists.txt
index 6d5b0a2..7ad5cc0 100644
--- a/cmake/libpython/CMakeLists.txt
+++ b/cmake/libpython/CMakeLists.txt
@@ -467,6 +467,7 @@ function(add_libpython name type)
         LIBRARY DESTINATION ${LIBPYTHON_LIBDIR}
         RUNTIME DESTINATION ${LIBPYTHON_LIBDIR}
         PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Development
+        FRAMEWORK DESTINATION "Frameworks"
     )
 endfunction()
 
@@ -492,6 +493,18 @@ if(BUILD_SHARED)
         set_property(
             SOURCE ${SRC_DIR}/Modules/posixmodule.c
             APPEND PROPERTY COMPILE_DEFINITIONS WITH_NEXT_FRAMEWORK)
+
+        if(BUILD_FRAMEWORK)
+            set_target_properties(libpython-shared PROPERTIES
+                FRAMEWORK TRUE
+                FRAMEWORK_VERSION "${PY_VERSION_MAJOR}.${PY_VERSION_MINOR}"
+                MACOSX_FRAMEWORK_IDENTIFIER org.python.python
+                MACOSX_FRAMEWORK_SHORT_VERSION_STRING "${PY_VERSION_MAJOR}.${PY_VERSION_MINOR}"
+                MACOSX_FRAMEWORK_BUNDLE_VERSION "${PY_VERSION_MAJOR}.${PY_VERSION_MINOR}"
+                XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
+                OUTPUT_NAME Python
+            )
+        endif()
     endif()
 
     set(targetname "libpython3-shared")
-- 
2.5.5

