From f5994e8fb1b5d021d3e1cec0e789188e25e7bf1d Mon Sep 17 00:00:00 2001
From: Marcel Metz <mmetz@adrian-broher.net>
Date: Sat, 9 Apr 2016 20:16:24 +0200
Subject: [PATCH 5/5] Add Mac OSX framework support

---
 CMakeLists.txt                 | 10 ++++++++++
 cmake/libpython/CMakeLists.txt | 13 +++++++++++++
 2 files changed, 23 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 05aafee..5e55a89 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -48,6 +48,16 @@ else()
   set(BUILD_WININST OFF)
   set(BUILD_WININST_ALWAYS OFF)
 endif()
+if(APPLE)
+  option(BUILD_FRAMEWORK "Build a Mac OSX Framework" OFF)
+  if(BUILD_FRAMEWORK)
+    set(BUILD_SHARED ON)
+    set(BUILD_STATIC OFF)
+  endif()
+else()
+  set(BUILD_FRAMEWORK OFF)
+endif()
+
 
 # Append an item to a string if it does not already contain it.  Any additional
 # arguments are treated as extra "set" arguments
diff --git a/cmake/libpython/CMakeLists.txt b/cmake/libpython/CMakeLists.txt
index 6d5b0a2..525baa2 100644
--- a/cmake/libpython/CMakeLists.txt
+++ b/cmake/libpython/CMakeLists.txt
@@ -462,11 +462,24 @@ function(add_libpython name type)
     # Export target
     set_property(GLOBAL APPEND PROPERTY PYTHON_TARGETS ${name})
 
+    if(BUILD_FRAMEWORK)
+        set_target_properties(${name} PROPERTIES
+            FRAMEWORK TRUE
+            FRAMEWORK_VERSION "${PY_VERSION_MAJOR}.${PY_VERSION_MINOR}"
+            MACOSX_FRAMEWORK_IDENTIFIER org.python.python
+            MACOSX_FRAMEWORK_SHORT_VERSION_STRING "${PY_VERSION_MAJOR}.${PY_VERSION_MINOR}"
+            MACOSX_FRAMEWORK_BUNDLE_VERSION "${PY_VERSION_MAJOR}.${PY_VERSION_MINOR}"
+            XCODE_ATTRIBUTE_INSTALL_PATH "@rpath"
+            OUTPUT_NAME Python
+        )
+    endif()
+
     install(TARGETS ${name}
         ARCHIVE DESTINATION ${LIBPYTHON_ARCHIVEDIR}
         LIBRARY DESTINATION ${LIBPYTHON_LIBDIR}
         RUNTIME DESTINATION ${LIBPYTHON_LIBDIR}
         PUBLIC_HEADER DESTINATION ${INCLUDE_INSTALL_DIR} COMPONENT Development
+        FRAMEWORK DESTINATION "Frameworks"
     )
 endfunction()
 
-- 
2.5.5

